modules/runner/src/main/java/org/apache/ignite/internal/app/LifecycleManager.java

```diff
@@ -148,12 +153,11 @@ private synchronized void stopAllComponents() {
             }
         });
 
-        new ReverseIterator<>(startedComponents).forEachRemaining(component -> {
-            try {
-                component.stop();
-            } catch (Exception e) {
-                LOG.warn("Unable to stop component [component={}, nodeName={}]", e, component, nodeName);
-            }
-        });
+        List<CompletableFuture<Void>> allComponentsStopFuture = new ArrayList<>();
+
+        new ReverseIterator<>(startedComponents).forEachRemaining(component -> allComponentsStopFuture.add(component.stopAsync()));
+
+        allOf(allComponentsStopFuture.toArray(CompletableFuture[]::new))
+                .whenComplete((v, e) -> stopFuture.complete(null));
```

Comment:
I'd say that we should explicitly specify pool to execute start/stop, otherwise we may switch thread unpredictably.